app/src/main/java/com/example/test/UserManager.kt

```
@@ -0,0 +1,46 @@
+package com.example.test
+
+import android.content.Context
+import android.database.Cursor
+
+class UserManager(val context:Context){
+
+    fun addUser(user: User){
+        val dataBaseHelper = DataBaseHelper(context)
+        dataBaseHelper.addData(user.email,user.fName,user.lName,user.password)
+    }
+
+    fun fetchEmails(): ArrayList<String>{
+        val dataBaseHelper = DataBaseHelper(context)
+        var emails = ArrayList<String>()
+        var data = dataBaseHelper.fetchData()
+        while (data.moveToNext()){
+            emails.add(data.getString(0))
+        }
+        return emails
+    }
+
+    fun isAvailable(email: String?):Boolean{
+        val list = fetchEmails()
+        for(mail in list){
+            if(email == mail)return true
+        }
+        return false
+    }
+
+    fun fetchUser(email : String) : User{
```

If the user doesn't exist the you propagate this to the upper components instead of failing silently by returning an empty user.